src/pages/smart-contracts/setting-up-local-environment.md

```diff
@@ -0,0 +1,332 @@
+---
+title: Setting up local environment
+description: Learn how to setup a local smart contract development environment
+duration: 45 minutes
+experience: advanced
+---
+
+## Introduction
+
+As soon as you get to work on your smart conrtacts and testing them end to end within your application, you need a local development environment that supports all steps and helps you achieve faster iteration cycles. This is critical because transaction times on the testnet take a few minutes and would slow you down. The solution is to run a local version of the testnet, often called a "mocknet".
+
+By the end of this tutorial, you will:
+
+- Have all required tooling and services up and running locally
+- Be able to improve and re-deploy smart contracts quickly
+- Test your smart contract inside your app
+
+## Prerequisites
+
+### Prior tutorials
+
+You need to be familiar with Clarity and the Stacks authentication (using Connect). Please finish all previous tutorials, especially the [Counter tutorial](/smart-contracts/counter-tutorial) and the [Building a todo app tutorial](/authentication/building-todo-app).
+
+### Environment
+
+You should have [NodeJS](https://nodejs.org/en/download/) and [Docker](https://docs.docker.com/get-docker/) installed on your workstation.
+
+## Overview
+
+For a complete local mocknet setup, you need the following services and tools:
+
+- [Stacks Node](https://github.com/blockstack/stacks-blockchain)
+- [Stacks Blockchain API](https://github.com/blockstack/stacks-blockchain-api)
+- [Stacks Explorer](https://github.com/blockstack/explorer)
+- [Stack Authenticator](https://github.com/blockstack/ux/tree/master/packages/app)
+- [Todo's web with Connect](https://github.com/blockstack/blockstack-todos/)
+
+## Step 1: Set up Stacks Node and Blockchain API
+
+The easiest way to spin up both of these locally is by booting up a docker image:
+
+```bash
+docker run -p 3999:3999 blockstack/stacks-blockchain-api-standalone mocknet
+```
+
+The initial start will take a few minutes. As soon as it is complete, you should be able to open up `http://localhost:3999/` and see:
+
+```js
+{"status":"ready"}
+```
+
+## Step 2: Set up Stacks Explorer
+
+The Stacks Explorer makes it easy way to review transactions. Let's spin up a local version:
+
+```bash
+git clone https://github.com/blockstack/explorer.git
```

Comment:
@wileyj could we also just use a docker image for this project?

Reply:
yes! we have an image that we use for the prod explorer. working on a fully-dockerized instruction set. 
the image is here: `quay.io/blockstack/explorer:latest`